packages/dd-trace/src/plugins/ci_plugin.js

```diff
@@ -91,6 +91,13 @@ module.exports = class CiPlugin extends Plugin {
           ...testModuleSpanMetadata
         }
       })
+      // only for vitest
```

Comment:
Does the order matter? I mean, could we add this in `datadog-plugin-vitest/src/index.js` this in the constructor? Only to prevent vitest specific code in general CiPlugin file.

```javascript
this.addSub(`ci:vitest:session:start`, ({ command, frameworkVersion, rootDir }) => {
    process.env.DD_CIVISIBILITY_TEST_SESSION_ID = this.testSessionSpan.context().toTraceId()
    process.env.DD_CIVISIBILITY_TEST_MODULE_ID = this.testModuleSpan.context().toSpanId()
}
```

Reply:
the constructor is not a good place for that because `this.testSessionSpan` and `this.testModuleSpan` are not defined yet, and I can't start those spans there either, because I want them to start when the test session actually starts, not at the point where `dd-trace` is initialized.

I think I can do something like though:

```javascript
class CiPlugin {
   this.addSub(`ci:${this.constructor.id}:session:start`, this.sessionStartHandler.bind(this))
   sessionStartHandler () {
      // current code
   }
}
...
class VitestPlugin {
  this.addSub(`ci:${this.constructor.id}:session:start`, function () {
     this.sessionStartHandler.apply(this, arguments)
     process.env.DD_CIVISIBILITY_TEST_SESSION_ID = this.testSessionSpan.context().toTraceId()
     process.env.DD_CIVISIBILITY_TEST_MODULE_ID = this.testModuleSpan.context().toSpanId()
  })
}
```
WDYT?

Reply:
the constructor is not a good place for that because `this.testSessionSpan` and `this.testModuleSpan` are not defined yet, and I can't start those spans there either, because I want them to start when the test session actually starts, not at the point where `dd-trace` is initialized. 

I could do something like:

```javascript
  class CiPlugin {
    ...
    this.addSub(`ci:${this.constructor.id}:session:start`, function () {
      // handled by VitestPlugin
      if (this.constructor.id === 'vitest') {
        return
      }
      this.getSessionStartHandler.apply(this, arguments)
    })
    getSessionStartHandler ({ command, frameworkVersion, rootDir }) {
       // current code
    }
  ...

  class VitestPlugin extends CiPlugin {
    ...
    this.addSub(`ci:vitest:session:start`, function () {
      this.getSessionStartHandler.apply(this, arguments)
      process.env.DD_CIVISIBILITY_TEST_SESSION_ID = this.testSessionSpan.context().toTraceId()
      process.env.DD_CIVISIBILITY_TEST_MODULE_ID = this.testModuleSpan.context().toSpanId()
    })
  }
```

`CiPlugin` still has to know about vitest but at least there's no other specific logic. It's not great but it might be better? WDYT? Or can you think of other alternatives?

Reply:
I hadn't completely understood your comment, my bad. Indeed the order does matter: `this.testSessionSpan` needs to be defined, so the parent handler needs to be called first. I don't think we can guarantee the order so it'll have to stay that way sadly

Reply:
if order matters, no changes needed. Ok if you want to call to noop method by default that is overridden in VitestPlugin, or leave it as it is now